coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/attachment/ProposalDevelopmentAttachmentController.java

```diff
@@ -392,9 +393,12 @@ public ModelAndView getProposalAttachmentRights(@ModelAttribute("KualiForm") Pro
 
         List<NarrativeUserRights> editableRights = new ArrayList<NarrativeUserRights>();
         for (NarrativeUserRights right : selectedNarrative.getNarrativeUserRights()) {
-            NarrativeUserRights editableRight = new NarrativeUserRights();
-            PropertyUtils.copyProperties(editableRight,right);
-            editableRights.add(editableRight);
+            if (getKraAuthorizationService().hasPermission(right.getUserId(),form.getProposalDevelopmentDocument(), PermissionConstants.MODIFY_NARRATIVE) ||
+                    getKraAuthorizationService().hasPermission(right.getUserId(),form.getProposalDevelopmentDocument(), PermissionConstants.VIEW_NARRATIVE) ) {
+                NarrativeUserRights editableRight = new NarrativeUserRights();
+                PropertyUtils.copyProperties(editableRight,right);
+                editableRights.add(editableRight);
```

Comment:
I feel like this isn't completely correct. This will just correct the view of the permissions, but the narrative user rights value will still exist even though the user shouldn't have permission right? Seems like the user's NarrativeRights should be removed at the same time they are removed from having access to the proposal, instead of just hiding their permissions from view.